core/src/main/scala/algebra/ring/Ring.scala

```diff
@@ -32,11 +32,11 @@ trait Ring[@sp(Int, Long, Float, Double) A] extends Any with Rig[A] with Rng[A]
   def fromInt(n: Int): A = sumN(one, n)
 }
 
-trait RingFunctions {
+trait RingFunctions extends AdditiveGroupFunctions with MultiplicativeMonoidFunctions {
```

Comment:
in the later case you wan write `Ring.fromInt(x)` and use type inference. In the former you can't.

But yeah, there is an argument to be made that we should just live with the version where we write the type explicitly.